etcdserver/api/v3rpc/watch.go

```diff
@@ -277,29 +277,13 @@ func (sws *serverWatchStream) recvLoop() error {
 				creq.RangeEnd = []byte{}
 			}
 
-			err := sws.isWatchPermitted(creq)
-			if err != nil {
-				var cancelReason string
-				switch err {
-				case auth.ErrInvalidAuthToken:
-					cancelReason = rpctypes.ErrGRPCInvalidAuthToken.Error()
-				case auth.ErrAuthOldRevision:
-					cancelReason = rpctypes.ErrGRPCAuthOldRevision.Error()
-				case auth.ErrUserEmpty:
-					cancelReason = rpctypes.ErrGRPCUserEmpty.Error()
-				default:
-					if err != auth.ErrPermissionDenied {
-						sws.lg.Error("unexpected error code", zap.Error(err))
-					}
-					cancelReason = rpctypes.ErrGRPCPermissionDenied.Error()
-				}
-
+			if !sws.isWatchPermitted(creq) {
```

Comment:
I think we can keep the server side change?

Reply:
ah sorry, let me open a follow up PR

Reply:
On the second thought, I guess just reverting the entire change might be safer? What do you think?

Reply:
Actually we kept the server side change when reverting the PR on 3.5.  Keeping the server side change can provide more useful error message (`cancelReason`).

But it isn't a big deal, it's up to you. thx

Reply:
got it, opened https://github.com/etcd-io/etcd/pull/15584